pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -67,20 +75,36 @@ public boolean supportSchemaVersioning() {
 
     @Override
     public GenericRecord decode(byte[] bytes, byte[] schemaVersion) {
-        ensureSchemaInitialized();
-
+        if (schema == null) {
+            try {
+                SchemaInfo schemaInfo = schemaInfoProvider.getLatestSchema().get();
+                schema = generateSchema(schemaInfo);
+                ensureSchemaInitialized();
+                schema.setSchemaInfoProvider(schemaInfoProvider);
+                    log.info("Configure {} schema for topic {} : {}",
+                            componentName, topicName, schemaInfo.getSchemaDefinition());
+            } catch (InterruptedException | ExecutionException e ) {
```

Comment:
If it is InterruptedException, you should propagate the interrupted signal through `Thread.currentThread().interrupt()`.